platform/core/PolyPodCoreSwift/Sources/PolyPodCoreSwift/UpdateNotification.swift

```diff
@@ -1,87 +1,43 @@
-import UIKit
-
-public protocol UpdateNotificationStore {
-    func read() -> (Int, String)?
-    func write(id: Int, state: String)
-}
+import Foundation
 
 public class UpdateNotification {
-    private enum State: String, CaseIterable {
-        case NOT_SEEN
-        case PUSH_SEEN
-        case ALL_SEEN
-
-        static func parse(_ s: String?) -> State? {
-            guard let s = s else { return nil }
-            for value in allCases where value.rawValue == s {
-                return value
-            }
-            return nil
-        }
-    }
-
-    private static var id: Int?
-    private static var store: UpdateNotificationStore?
-    private static var cachedState: State?
-    private static var state: State {
-        get { cachedState! }
-        set {
-            cachedState = newValue
-            store!.write(id: id!, state: cachedState!.rawValue)
-        }
-    }
-
-    private static func loadLastState() -> State {
-        if id == 0 {
-            return .ALL_SEEN
-        }
-
-        guard let (lastId, lastState) = Self.store!.read() else {
-            return .NOT_SEEN
-        }
-
-        if id! < lastId {
-            return .ALL_SEEN
-        }
-
-        if id != lastId {
-            return .NOT_SEEN
-        }
-
-        return State.parse(lastState) ?? .ALL_SEEN
-    }
-
-    public static func initialize(id: Int, store: UpdateNotificationStore) {
-        self.id = id
-        self.store = store
-        cachedState = Self.loadLastState()
-    }
-
     public static var showPush: Bool {
-        return state == .NOT_SEEN
+        let result: Result<Bool, Error> = Core.instance.executeRequest(.getShowPushNotification)
+        switch result {
+        case .success(let value):
+            return value
+        case .failure(let error):
+            print("getShowPushNotification request failed: \(error.localizedDescription)")
```

Comment:
We don't have `Log` in the Core package, so I went for a good old print in case any of these requests fail. But:

1. I don't think that would actually show up in the logs - would it?
2. Seems like an unnecessary amount of boilerplate - maybe we want to have some default error handling in `executeRequest` that logs the error, so that we don't need to repeat this wherever we call it?

Reply:
1. You should use `Log.error`.
2. Yeah, we can do that. It was not done with the intent that the error will be logged as higher as possible.